gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -124,32 +124,10 @@ case class OffloadJoin() extends OffloadSingleNode with LogLevelUtil {
     }
     plan match {
       case plan: ShuffledHashJoinExec =>
-        val left = plan.left
-        val right = plan.right
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
-        BackendsApiManager.getSparkPlanExecApiInstance
-          .genShuffledHashJoinExecTransformer(
-            plan.leftKeys,
-            plan.rightKeys,
-            plan.joinType,
-            TransformHints.getShuffleHashJoinBuildSide(plan),
-            plan.condition,
-            left,
-            right,
-            plan.isSkewJoin)
+        ShuffledHashJoinExecTransformerBase.from(plan)
       case plan: SortMergeJoinExec =>
```

Comment:
We should also change to use shuffled hash join transformer to do validate when add sort merge join hint. Otherwise, the shj may fail because it only pass the smj validation. But it's a bit confused to add a shj hint to smj, we may want a smj transformer at the end.

Reply:
Do validate is in `AddTransformHintRule`
```
val transformer = OffloadJoin.transformSortMergeJoinExec(plan)
transformer.doValidate().tagOnFallback(plan)
```

Reply:
I think the expected behavior for the smj is:
1. convert smj to shj if can offload shj and enable the config, otherwise 2
2. offload smj, otherwise 3
3. fallback smj

So, I think we miss the part 2: if we can not offload shj but support offload smj.

Reply:
OffloadJoin.transformSortMergeJoinExec(smj) Will determine whether to offload smj or shj based on conditions.

Reply:
I think it does not help at least in these two cases:
1. Do we support offload smj if the shj validation failed ?
2. What happens if we disable offload smj, `spark.gluten.sql.columnar.sortMergeJoin=false`? before we can offload shj.

Reply:
There is indeed a problem, thank you, I will fix it right away.

Reply:
Regarding the first point, how about we keep the logic before PR, when force shj, if validation fails, fallback?

Reply:
fixed.